client/comms/wsconn.go

```diff
@@ -212,8 +227,8 @@ func (conn *wsConn) connect(ctx context.Context) error {
 	}
 	ws, _, err := dialer.DialContext(ctx, conn.cfg.URL, nil)
 	if err != nil {
-		var e x509.UnknownAuthorityError
-		if errors.As(err, &e) {
+		var e x509.HostnameError // No need to retry...
+		if IsErrorInvalidCert(err) || errors.As(err, &e) {
```

Comment:
Why isn't `var e x509.HostnameError` part of `IsErrorInvalidCert`?  It's not actually saying there's a problem with the host name but that it doesn't match the cert.  Hence why it is a cert error defined by the `x509` package.

Reply:
e.g. https://github.com/nats-io/nats-server/pull/1264

Reply:
updated `IsErrorInvalidCert` in my latest push.